publisher/publisher.go

```diff
@@ -67,6 +69,11 @@ type Impl struct {
 	SA core.StorageAuthority
 }
 
+// as a gRPC server without changing its AMQP public API.
+type GRPCWrapper struct {
+	Inner *Impl
```

Comment:
I believe Inner doesn't need to be exported here either. You can just define a function in this package that returns the appropriate type.